### PR TITLE
fix: Reload registry TLS certificate when it is updated on the file system

### LINKED ISSUE #803: :bug: To update the TLS certificate used by the OCI registry (`serve bundle` command), the process must be restarted
This is an issue when the server certificate is regularly rotated, but the process is not regularly restarted. This is common when mindthegap runs in a Pod.

### PR DESCRIPTION
**Pull request by dlipovetsky**
Watch the filesystem for changes to the certificate, and reload when it is updated.

Introduces a dependency on https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/certwatcher. This does not appear to impact the executable size.

The first commit is a test. On its own, it fails. With the fix in the second commit, the test passes.

Fixes #803

(Note: This is stacked on #804)

### CODE REVIEW

File: docker/registry/registry.go [hunk 1]
```diff
@@ -146,10 +147,28 @@ func (r Registry) Shutdown(ctx context.Context) error {
...
+		if cwErr != nil {
+			return fmt.Errorf("failed to read TLS certificate or key: %w", cwErr)
+		}
+		r.delegate.TLSConfig = &tls.Config{
+			GetCertificate: watcher.GetCertificate,
+			MinVersion:     tls.VersionTLS12,
+		}
+		go func() {
+			if startErr := watcher.Start(context.TODO()); startErr != nil {
+				log.Error(startErr, "certwatcher Start failed")
```
Comment (dkoshkin) [inline docker/registry/registry.go:165]: Do we not want to error here?

File: docker/registry/registry.go [hunk 2]
```diff
@@ -146,10 +147,28 @@ func (r Registry) Shutdown(ctx context.Context) error {
...
 }
 
-func (r Registry) ListenAndServe() error {
+func (r Registry) ListenAndServe(log logr.Logger) error {
+	log = log.WithName("registry")
+
 	var err error
 	if r.config.HTTP.TLS.Certificate != "" && r.config.HTTP.TLS.Key != "" {
-		err = r.delegate.ListenAndServeTLS(r.config.HTTP.TLS.Certificate, r.config.HTTP.TLS.Key)
+		watcher, cwErr := certwatcher.New(r.config.HTTP.TLS.Certificate, r.config.HTTP.TLS.Key)
```
Comment (dkoshkin) [inline docker/registry/registry.go:155]: Nice use of this utility!

**Review by dkoshkin — DISMISSED** [review]
Thanks for the separate commits and messages, made it much simpler to understand the changes.

File: docker/registry/registry.go (hunk 1 above)
Comment (jimmidyson) [reply docker/registry/registry.go]: Added a panic here.

**Review by jimmidyson — APPROVED** [review]

**Comment by jimmidyson** [comment]
Thank you!

**Pull request merged**